From 5f66992c37637c87c8ef56181ea4807747fc4f93 Mon Sep 17 00:00:00 2001
From: Lukas Bulwahn <lukas.bulwahn@gmail.com>
Date: Sun, 14 Apr 2019 13:09:29 +0200
Subject: [PATCH 14/15] freezer: reorder to be in scope; make static (breaking
 it?)

kernel/cgroup/freezer.c:46:29: error: use of undeclared identifier 'freezer_mutex'; did you mean 'freeze_super'?
} __attribute__((guarded_by(freezer_mutex)));
                            ^~~~~~~~~~~~~
                            freeze_super
./include/linux/fs.h:2297:12: note: 'freeze_super' declared here
extern int freeze_super(struct super_block *super);
           ^
kernel/cgroup/freezer.c:46:18: warning: 'guarded_by' attribute only applies to non-static data members and global variables [-Wignored-attributes]
} __attribute__((guarded_by(freezer_mutex)));
                 ^
1 warning and 1 error generated.

This freezer mutex was not a good example.

I misread the warning, it supposed to be non-static and global.

Clang reported after this change:

kernel/cgroup/freezer.c:49:18: warning: 'guarded_by' attribute only applies to non-static data members and global variables [-Wignored-attributes]
} __attribute__((guarded_by(freezer_mutex)));
                 ^
kernel/cgroup/freezer.c:46:1: warning: 'static' ignored on this declaration [-Wmissing-declarations]
static struct freezer {
^

Okay, so it complains about static and then says afterwards that static is ignored anyway.
---
 kernel/cgroup/freezer.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/kernel/cgroup/freezer.c b/kernel/cgroup/freezer.c
index 0bbd038ca04d..ab4b27464eca 100644
--- a/kernel/cgroup/freezer.c
+++ b/kernel/cgroup/freezer.c
@@ -40,13 +40,14 @@ enum freezer_state_flags {
 	CGROUP_FREEZING		= CGROUP_FREEZING_SELF | CGROUP_FREEZING_PARENT,
 };
 
-struct freezer {
+static DEFINE_MUTEX(freezer_mutex);
+
+// could this be made static?? I would need to get what this really does here.
+static struct freezer {
 	struct cgroup_subsys_state	css;
 	unsigned int			state;
 } __attribute__((guarded_by(freezer_mutex)));
 
-static DEFINE_MUTEX(freezer_mutex);
-
 static inline struct freezer *css_freezer(struct cgroup_subsys_state *css)
 {
 	return css ? container_of(css, struct freezer, css) : NULL;
-- 
2.17.1

